tests_e2e/tests/ext_cgroups/install_extensions.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python3
+
+# Microsoft Azure Linux Agent
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from datetime import datetime, timedelta
+from pathlib import Path
+
+from tests_e2e.tests.lib.agent_test import AgentTest
+from tests_e2e.tests.lib.agent_test_context import AgentTestContext
+from tests_e2e.tests.lib.identifiers import VmExtensionIds
+from tests_e2e.tests.lib.logging import log
+from tests_e2e.tests.lib.ssh_client import SshClient
+from tests_e2e.tests.lib.virtual_machine_extension_client import VirtualMachineExtensionClient
+
+
+class InstallExtensions(AgentTest):
+    """
+    This test installs the multiple extensions on verify they are running in correct cgroups.
+    """
+
+    def __init__(self, context: AgentTestContext):
+        super().__init__(context)
+        self._ssh_client = SshClient(
+            ip_address=self._context.vm_ip_address,
+            username=self._context.username,
+            private_key_file=self._context.private_key_file)
+
+    def run(self):
+        self._prepare_agent()
+        # Install the GATest extension
+        self._install_gatest_extension()
+        # Install the Azure Monitor Agent
+        self._install_ama()
+        # Install the VM Access extension
+        self._install_vmaccess()
+        # Install the CSE extension
+        self._install_cse()
+
+    def _prepare_agent(self):
+        log.info("=====Updating monitoring deadline for tracking azuremonitoragent service=====")
+        future_date = datetime.utcnow() + timedelta(days=2)
+        expiry_time = future_date.date().strftime("%Y-%m-%d")
+        # We hardcoded the service name to monitor for sometime in production without need of manifest update.
+        # we no longer monitoring by default in prod but using it in our test validation.
+        result = self._ssh_client.run_command(f"update-waagent-conf Debug.CgroupMonitorExpiryTime={expiry_time}", use_sudo=True)
```

Comment:
I don't fully understand this comment, probably because I don't have a good understanding of this extension. 

What do you mean we hardcoded the service name? And what do you mean by monitoring in prod?

Reply:
added more context, let me know if it's still not clear